core/worker.go

```diff
@@ -916,7 +917,7 @@ func (w *worker) FinalizeAssemble(chain consensus.ChainHeaderReader, header *typ
 	manifestHash := w.ComputeManifestHash(parent.Header())
 	block.Header().SetManifestHash(manifestHash)
 
-	if nodeCtx == common.ZONE_CTX {
+	if nodeCtx == common.ZONE_CTX && w.hc.ProcessingState() {
```

Comment:
Can a zone headerchain create a valid pending header?

Reply:
no it cannot, we cannot compute anything to do with the state, i.e state root, ext root, tx root, receipt root

Reply:
So if nodeCtx == common.ZONE_CTX but w.hc.ProcessingState() == false this is an invalid state right? We should log an error because this function shouldn't be called in that case